quickstarts/microsoft.network/application-gateway-waf-firewall-policy/main.bicep

```diff
@@ -0,0 +1,516 @@
+        backendAddressPool: contains(rule, 'backendPool') ? json('{"id": "${resourceId('Microsoft.Network/applicationGateways/backendAddressPools', applicationGatewayName, rule.backendPool)}"}') : null
+        backendHttpSettings: contains(rule, 'backendHttpSettings') ? json('{"id": "${resourceId('Microsoft.Network/applicationGateways/backendHttpSettingsCollection', applicationGatewayName, rule.backendHttpSettings)}"}') : null
+        redirectConfiguration: contains(rule, 'redirectConfiguration') ? json('{"id": "${resourceId('Microsoft.Network/applicationGateways/redirectConfigurations', applicationGatewayName, rule.redirectConfiguration)}"}') : null
+      }
+    }]
+    redirectConfigurations: [for redirectConfiguration in redirectConfigurations: {
+      name: redirectConfiguration.name
+      properties: {
+        redirectType: redirectConfiguration.redirectType
+        targetUrl: redirectConfiguration.targetUrl
+        targetListener: contains(redirectConfiguration, 'targetListener') ? json('{"id": "${resourceId('Microsoft.Network/applicationGateways/httpListeners', applicationGatewayName, redirectConfiguration.targetListener)}"}') : null
+        includePath: redirectConfiguration.includePath
+        includeQueryString: redirectConfiguration.includeQueryString
+        requestRoutingRules: [
+          {
+            id: resourceId('Microsoft.Network/applicationGateways/requestRoutingRules', applicationGatewayName, redirectConfiguration.requestRoutingRule)
+          }
+        ]
+      }
+    }]
+  }
+}
+
+resource applicationGatewayDiagnostics 'microsoft.insights/diagnosticSettings@2021-05-01-preview' = if (enableDiagnostics) {
+  scope: applicationGateway
+  name: appGatewayDiagnosticsName
+  properties: {
+    workspaceId: empty(logAnalyticsWorkspaceId) ? null : logAnalyticsWorkspaceId
+    storageAccountId: empty(diagnosticStorageAccountId) ? null : diagnosticStorageAccountId
+    logs: [
+      {
+        category: 'ApplicationGatewayAccessLog'
+        enabled: true
+      }
+      {
+        category: 'ApplicationGatewayPerformanceLog'
+        enabled: true
+      }
+      {
+        category: 'ApplicationGatewayFirewallLog'
+        enabled: true
+      }
+    ]
+    metrics: [
+      {
+        category: 'AllMetrics'
+        enabled: true
+      }
+    ]
+  }
+}
+
+resource applicationGatewayLock 'Microsoft.Authorization/locks@2017-04-01' = if (enableDeleteLock) {
+  scope: applicationGateway
+  name: appGatewayLockName
+  properties: {
+    level: 'CanNotDelete'
+  }
+}
+
+resource firewallPolicy 'Microsoft.Network/ApplicationGatewayWebApplicationFirewallPolicies@2021-03-01' = if (enableWebApplicationFirewall) {
+  name: firewallPolicyName == '' ? 'placeholdervalue' : firewallPolicyName // placeholder value required as name cannot be empty/null when enableWebApplicationFirewall equals false 
```

Comment:
is this still needed?  I think we might have deployed a fix for it...

Reply:
@bmoore-msft i believe it is still needed. ive tried the following combinations and all resort with the error ```The name property cannot be null or empty```

```name: firewallPolicyName != '' ? firewallPolicyName : ''```
```name: !empty(firewallPolicyName) ? firewallPolicyName : ''```
```name: firewallPolicyName``` 

im running on latest bicep version ```0.4.1124```